shcomplete/tfdf.py

```diff
@@ -0,0 +1,132 @@
+import os
+import re
+import logging
+import pickle
+import numpy
+import pygtrie
+
+from math import log
+from clint.textui import progress
+
+
+def get_tries(args):
+    """
+    Return a list of tries, one for each file where the keys are the command lines.
+    """
+    log = logging.getLogger("get_tries")
+    tries = []
+    max_length = args.max_length
+    for root, _, files in os.walk(args.directory):
+        for file_name in files:
+            path_to_file = os.path.join(root, file_name)
+            log.info("parsing %s", path_to_file)
+            trie = pygtrie.StringTrie(separator=" ")
+            tries.append((path_to_file, trie))
+            with open(path_to_file, "r") as f:
+                for line in f:
+                    line = [l for l in line.rstrip().replace("\t", " ").split(" ")][:max_length]
+                    key = " ".join(line)
+                    if key not in trie:
+                        trie[key] = 0
+                    node = trie._root
+                    prefixes = []
+                    for i in range(1, len(line)+1):
+                        prefixes.append(" ".join(line[:i]))
+                    for pfx in prefixes:
+                        try:
+                            trie[pfx] += 1
+                        except KeyError:
+                            trie[pfx] = 1
```

Comment:
I know this is not the best way of doing it, but I don't know how getting node.values of the prefixes with children since children returned the nodes of single tokens.
Anyway this was not the bottleneck, it runs fast, so I let it like this for the moment